integration/framework.go

```diff
@@ -293,9 +295,60 @@ func (f *framework) deleteGuestCluster() error {
 		return microerror.Mask(err)
 	}
 
+	// TODO: during the cloudformation migration the legacy resource is always deleted last,
+	// when the migration is done we will need to check here the cloudformation stack deletion
+	// message
 	logEntry := "cluster '${CLUSTER_NAME}' deleted"
-	if os.Getenv("VERSION_BUNDLE_VERSION") == "0.2.0" {
-		logEntry = "deleting AWS cloudformation stack: deleted"
-	}
 	return f.waitForPodLog("giantswarm", logEntry, operatorPodName)
 }
+
+func (f *framework) initGuestClientset() error {
```

Comment:
I'd wrap this in https://golang.org/pkg/sync/#Once and call every time in `apiUp`. Otherwise it requires to much knowledge outside the objects how to use it.

I also tend to export the methods which are used outside the object even if it's private. I.e. always think about the object as it could be exported one day by just changing the first letter. I think it makes it easier to maintain that way.

Reply:
> I'd wrap this in https://golang.org/pkg/sync/#Once and call every time in apiUp. Otherwise it requires to much knowledge outside the objects how to use it.

I agree with the encapsulation of inner workings, but we don't need to use `sync.Once` here, just checking for the zero value of the clientset is enough IMO, let me know WDYT.

> I also tend to export the methods which are used outside the object even if it's private. I.e. always think about the object as it could be exported one day by just changing the first letter. I think it makes it easier to maintain that way.

Sure, maybe it's still too early but fine with me.

Reply:
> but we don't need to use sync.Once here, just checking for the zero value of the clientset is enough IMO

100% I missed that detail.